zerver/tests/test_message_fetch.py

```diff
@@ -584,10 +584,9 @@ def test_add_term_using_in_operator(self) -> None:
         self._do_add_term_test(term, "WHERE (recipient_id NOT IN (__[POSTCOMPILE_recipient_id_1]))")
 
     def test_add_term_using_in_operator_and_negated(self) -> None:
-        # negated = True should not change anything
```

Comment:
This comment is very puzzling, since it seems to suggest the previous behavior was intentional! 

Looks like it was added in e3e03e29460075ed1ec31961fab1ff1cd5ed8f28, so that suggests that unfortunately the contributor encountered this bug while writing those tests ... but incorrectly just wrote a test asserting the buggy behavior.